src/plugins/score-plugin-midi/Midi/MidiDrop.hpp

```diff
@@ -18,11 +18,101 @@ class DropHandler final : public Process::ProcessDropHandler
       const score::DocumentContext& ctx) const noexcept override;
 };
 
+struct PitchbendData{
+  int16_t bend;
+};
+struct ControllerData {
+  midi_size_t channel;
+  midi_size_t number;
+  midi_size_t value;
+  static ControllerData make_cc(const midi_size_t channel, const midi_size_t controller, const midi_size_t value)
+  {
+    SCORE_ASSERT(channel >= 0 && channel < 16);
+    return ControllerData{channel, controller, value};
+  }
+};
+
+struct NoteOnData {
+  midi_size_t channel;
+  midi_size_t note;
+  midi_size_t velocity;
+};
+struct NoteOffData {
+  midi_size_t channel;
+  midi_size_t note;
+  midi_size_t velocity;
+};
+struct MidiTrackEvent {
+  static MidiTrackEvent make_note_off(const double start, const midi_size_t ch, const midi_size_t n, const midi_size_t v){
```

Comment:
these functions look a lot like the ones in libremidi, maybe it should go there ?